augur/filter/include_exclude_rules.py

```diff
@@ -178,6 +190,10 @@ def filter_by_query(metadata, query) -> FilterFunctionReturn:
     set()
 
     """
+    # Try converting all queried columns to numeric.
+    for column in extract_variables(query).intersection(metadata.columns):
+        metadata[column] = pd.to_numeric(metadata[column], errors='ignore')
```

Comment:
In the force-pushes above, I split 77ef4a50a139a5b73272876e5488ecf8f417b1ad into b325b970d62517abcc0867766742ff915dc1012b + 2ead5b3e3306dc1100b49eb774287496018122d9.